Practical3-Completed CRUD Operation Practical.

Jay-Prajapat opened the pull request:
In this Practical I have implemented Product management system, In which I have implemented Add, Update, Edit product and also sort data by name and price in ascending and descending order and also implemented search function the can search product by id and name and show the products based on search.

Jay-Prajapat committed:
started implementing practical3

Jay-Prajapat committed:
Implemented insert data functionality.

Jay-Prajapat committed:
Implemented delete, edit, update and sort functionality.

Jay-Prajapat committed:
Search Functionality implemented.

Jay-Prajapat committed:
Completed CRUD Operation Practical.

Jay-Prajapat committed:
removed inline CSS from label.

Jay-Prajapat committed:
make changes in index.html for css file path and  change in script.js.

In script .js file when the local storage is empty it gives error and i have solved that error in show data function.

dilaara-simform commented on Practicals/Practical3/js/script.js:
```diff
@@ -0,0 +1,277 @@
+      element.prodId == id.value ||
+      element.name.toLowerCase().includes(id.value.toLowerCase())
+    ) {
+      console.log(element.prodId);
```
remove console.log

dilaara-simform commented on Practicals/Practical3/js/script.js:
```diff
@@ -0,0 +1,277 @@
+  let id = document.getElementById("prodId").value;
+
+  if (id == "") {
+    document.getElementById("pId").style.display = "block";
```
rather than repeating the lines you can create a common function and then use it.

Jay-Prajapat committed:
make changes in script.js file and index.html file

change max length validation for name,
implemented  when error the textbox should be highlighted red . Applied validation of specific format for product- Id and also removed console error.

Jay-Prajapat committed:
Make changes in UI part of the website.

Made website more attractive and user friendly.

Jay-Prajapat committed:
Created a common function named as changeStyle() in script.js file.

dilaara-simform commented on Practicals/Practical3/index.html:
```diff
@@ -0,0 +1,74 @@
+      </div>
+    </div>
+    <div class="search-container">
+      <input type="text" placeholder="Search by id or name..." name="search" id="search" />
```
filter should be at top of page.

dilaara-simform commented on Practicals/Practical3/index.html:
```diff
@@ -0,0 +1,74 @@
+    </div>
+    <form>
+      <div class="form-group">
+        <label for="prodId" class="label-name">Product ID:</label>
```
Improve form designing it should not cover the whole screen.

dilaara-simform commented on Practicals/Practical3/index.html:
```diff
@@ -0,0 +1,74 @@
+        <input type="file" class="form-control" id="image" />
+      </div>
+      <div class="form-group">
+        <label for="price" class="label-name">Price:</label>
```
price allows alphabet e to be entered please correct it.

dilaara-simform commented on Practicals/Practical3/index.html:
```diff
@@ -0,0 +1,74 @@
+        <textarea class="form-control" rows="5" id="description" maxlength="60"></textarea>
+      </div>
+      <button type="submit" class="btn btn-success" id="add" onclick="onSubmit()">
+        Add
```
onclick of this button nothing is visible in the listing.

Jay-Prajapat committed:
Make the changes that are given in the comment

Improved the design of  the screen, shifted the filter  at the top of the page, applied validation on image, in price validate the 'e' alphabet is not entered.
